pkg/resources/resource_grant_database.go

```diff
@@ -0,0 +1,139 @@
+package resources
+
+import (
+	"context"
+	"strings"
+
+	"github.com/MaterializeInc/terraform-provider-materialize/pkg/materialize"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/jmoiron/sqlx"
+)
+
+var grantDatabaseSchema = map[string]*schema.Schema{
+	"role_name": {
+		Description: "The name of the role to grant privilege to.",
+		Type:        schema.TypeString,
+		Required:    true,
+		ForceNew:    true,
+	},
+	"privilege": {
+		Description:  "The privilege to grant to the object.",
+		Type:         schema.TypeString,
+		Required:     true,
+		ForceNew:     true,
+		ValidateFunc: validPrivileges("DATABASE"),
+	},
+	"object": {
```

Comment:
Mentioned this on the Notion doc but figured it might get lost there. I feel like either we should follow Snowflake and make these top-level parameters...

```hcl
resource "materialize_grant_database" "..." {
    role_name = "..."
    privilege = "..."
    database_name = "..."
}
```

Or we should name the nested subresource after the object type rather than using `object`:

```hcl
resource "materialize_grant_database" "..." {
    role_name = "..."
    privilege = "..."
    database {
        name = "..."
    }
}
```